lib/matplotlib/backend_tools.py

```diff
@@ -318,7 +321,12 @@ class ToolQuit(ToolBase):
     default_keymap = rcParams['keymap.quit']
 
     def trigger(self, sender, event, data=None):
-        Gcf.destroy_fig(self.figure)
+        try:
+            manager = self.figure.canvas.manager
+        except:
```

Comment:
Bare exceptions catches everything, including Ctrl-C. I assume that you really want to capture AttributeError?